- Update measure error to include status, reason

linkous8 opened the pull request:
- Update measure error response to include info
- Update adjust error response to include reason

linkous8 committed:
- Update measure error to include status, reason
- Update measure error response to include info
- Update adjust error response to include reason

leokalev commented on servo:
```diff
@@ -97,8 +97,8 @@ def measure(param):
-        raise Exception('Measurement driver failed with status "{}" and message "{}"'.format(
-            status, rsp.get('message', 'undefined')))
+        raise DriverError('Measurement driver failed with status "{}" and message:\n{}'.format(
+            status, str(rsp.get('message', 'undefined'))), status=status, reason=rsp.get('reason', 'undefined'))
```
What is the reason for using str() here? The value of rsp['message'] is expected to be a string, anyway. NOTE that if an adjust/measure driver sends something that is not a string, the run_driver() function will fail (that's a separate bug, though, no action required for this commit)

linkous8 replied:
According to the relevant PR (https://github.com/opsani/servo/pull/10), response message was a type whose `__repr__` method produced undesirable results and needed to have the `__str__` method used instead. 

Specifically it calls out "python binary repr" but I was unable to contextualize this because it shouldn't be possible for message to be a byte str and binary numbers are stored in the standard int type whose `__repr__` returns base 10 even if instantiated with a binary literal.

@Johnius can you recall the adjust driver you were getting this behavior with? I just want to be sure removing the `str` invocation won't cause a regression

Johnius replied:
@linkous8 I can't say I ever had this issue at all. Also, aren't we moving to ServoX altogether?

linkous8 replied:
I can't speak to the timeline of moving to servox but I've been asked to make some changes to this model for fast fail. Thanks getting back on this, I prefer to understand the reason code was added before I go deleting it and this puts my mind at ease :+1:

leokalev commented on servo:
```diff
@@ -443,6 +443,9 @@ def exec_command():
         try:
             v = measure(p)
+        except DriverError as e:
+            traceback.print_exc()
```
This traceback seems redundant. DriverError is raised deliberately and doesn't benefit from a stack trace in the logs. My guess is that it was added here as a copy-paste from the 'ADJUST' case below. It isn't needed there, either - both can be safely removed.

linkous8 replied:
Good call, looks like the adjust case was copy pasted as well from the generic Exception handler where a traceback makes much more sense

linkous8 committed:
Remove needless traceback()s

linkous8 committed:
Remove unnecessary str cast

linkous8 committed:
Merge branch 'master' into feature/improve-error-reporting

linkous8 committed:
Fix base measure sending error message as reason
- Convert legacy status codes to status strings
- Don't pass in status arg when same as param default

benburdick reviewed (approved):
Tested and ready to merge

linkous8 commented:
Per @snarfbeest on Slack RE: 19dc5d65bc750b65337f07f6029bc89e139e51fa

>the changes are fine, I don't think this can create any problems in servo itself
